src/Main.cpp

```diff
@@ -114,13 +127,30 @@ static void SetLexer(const ScintillaGateway &editor, const std::string &language
 }
 
 static void CheckFileForNewLexer() {
+	std::string language;
+	
 	ScintillaGateway editor(GetCurrentScintilla());
 
-	if (config.over_ride || editor.GetLexer() == 1 /*SCLEX_NULL*/) {
+	wchar_t fileName[MAX_PATH] = { 0 };
+	SendMessage(nppData._nppHandle, NPPM_GETFULLCURRENTPATH, 0, (LPARAM)fileName);
+	std::string fileNameC = UTF8FromString(fileName).c_str();
+	const auto search = bufferLanguages.find(fileNameC);
+	if (search != bufferLanguages.end()) {
+		language = search->second;
+	}
+	else {
+		language = editor.GetLexerLanguage();
+		bufferLanguages[fileNameC] = language;
+	}
+
+	if (config.over_ride || (editor.GetLexer() == 1) /*SCLEX_NULL*/) {
```

Comment:
This should also check to see if `language` was set. For example

1. Open some file Notepad++ recognizes, such as .cpp
1. Set it to some other random language, such as pascal 
1. Switch to another file
1. Switch back. 

This should still keep the user's previous selection, however it gets re highlighted as whatever Notepad++ thinks it should be unless I'm understanding the functionality wrong.